src/go/k8s/pkg/resources/statefulset_test.go

```diff
@@ -87,6 +87,7 @@ func TestEnsure(t *testing.T) {
 			types.NamespacedName{},
 			types.NamespacedName{},
 			types.NamespacedName{},
+			types.NamespacedName{},
```

Comment:
meh, this is really terrible, we should improve it soon-ish :) (not your fault)

Reply:
Totally agree. 

I'm not sure why we have only one controller. We could leverage multiple controllers to managed parts of the resources: 

* loadbalancers
* superusers
* panda proxy
* kakfa listeners
* admin listeners
* archival
* metrics
etc

If we are feeling that we should stick to one controller, then better build pipeline of handler that will change resources based on very strict logic.